numba/tests/test_import.py

```diff
@@ -51,9 +51,63 @@ def test_laziness(self):
 
         out, _ = self.run_in_subproc(code)
         modlist = set(eval(out.strip()))
-        unexpected = set(blacklist) & set(modlist)
+        unexpected = set(banlist) & set(modlist)
         self.assertFalse(unexpected, "some modules unexpectedly imported")
 
+    def test_no_impl_import(self):
```

Comment:
@gmarkall thoughts on having a test like this for the CUDA target? Essentially something to make sure that `from numba import cuda` / `from numba.cuda import jit` doesn't trigger import of modules that are known to have registration side effects.

Reply:
Seems that `from numba import cuda` ends up importing `numba.cpython.cmathimpl` and `numba.cpython.mathimpl`.

Reply:
... though, this seems to be quite easily fixable.

Reply:
Test and fix added in https://github.com/numba/numba/commit/0611f058d611ad18276dae57734d4b53062d4180 - what do you think @stuartarchibald ?

Reply:
Thanks @gmarkall, the patch looks good. Are you happy for me to include it in this branch?

Reply:
Yes please!

Reply:
Great, thanks for confirming :)